__tests__/require-usecallback.test.ts

```diff
@@ -259,6 +259,41 @@ describe('Rule - Require-usecallback', () =>  {
         }`,
         errors: [{ messageId: "function-usecallback-props" }, { messageId: "function-usecallback-props" }],
       },
+      {
+        code: `
+        const Component = () => {
+          const myFn = async function test() {};
+          return <Child prop={myFn} />;
+        }`,
+        output: `import { useCallback } from 'react';
+
+        const Component = () => {
+          const myFn = useCallback(async () => {}, []);
+          return <Child prop={myFn} />;
+        }`,
+        errors: [{ messageId: "function-usecallback-props" }],
+      },
+      {
+        code: `
+        const Component = () => {
+          const myFn = async () => [];
+          return <Child prop={myFn} />;
+        }`,
+        output: `import { useCallback } from 'react';
+
+        const Component = () => {
+          const myFn = useCallback(async () => [], []);
+          return <Child prop={myFn} />;
+        }`,
+        errors: [{ messageId: "function-usecallback-props" }],
+      },
+      {
+        code: `
+        const Component = () => {
+          return <Child prop={async () => []} />;
+        }`,
+        errors: [{ messageId: "function-usecallback-props" }],
+      },
```

Comment:
Shouldn't the autofix create a prop like it does for the other cases?
```suggestion
      {
        code: `
        const Component = () => {
          return <Child prop={async () => []} />;
        }`,
         output: `import { useCallback } from 'react';

        const Component = () => {
        const prop = useCallback(async () => [], []);
          return <Child prop={prop} />;
        }`,
        errors: [{ messageId: "function-usecallback-props" }],
      },
```